core/lib/zksync_core/src/state_keeper/seal_criteria/mod.rs

```diff
@@ -177,6 +177,29 @@ impl IoSealCriteria for TimeoutSealer {
     }
 }
 
+#[derive(Debug, Clone, Copy)]
+pub(super) struct MiniblockMaxPayloadSizeSealer {
+    max_payload_size: usize,
+}
+
+impl MiniblockMaxPayloadSizeSealer {
+    pub fn new(config: &StateKeeperConfig) -> Self {
+        Self {
+            max_payload_size: config.miniblock_max_payload_size,
+        }
+    }
+}
+
+impl IoSealCriteria for MiniblockMaxPayloadSizeSealer {
+    fn should_seal_l1_batch_unconditionally(&mut self, _manager: &UpdatesManager) -> bool {
+        false
+    }
```

Comment:
Since this method doesn't really make sense for `MiniblockMaxPayloadSizeSealer`, I'd suggest to implement `should_seal_miniblock()` as an inherent method instead.

Reply:
Do you mean as a regular method implemented for `MempoolIO`?

Reply:
A regular method in `MiniblockMaxPayloadSizeSealer`.

Reply:
Fixed.